tree/tree/src/TTree.cxx

```diff
@@ -8524,27 +8553,42 @@ Int_t TTree::SetBranchAddress(const char* bname, void* addr, TClass* ptrClass, E
 /// Note: See the comments in TBranchElement::SetAddress() for the
 /// meaning of the addr parameter and the object ownership policy.
 
-Int_t TTree::SetBranchAddress(const char* bname, void* addr, TBranch** ptr, TClass* ptrClass, EDataType datatype, bool isptr)
+Int_t TTree::SetBranchAddress(const char *bname, void *addr, TBranch **ptr, TClass *ptrClass, EDataType datatype,
+                              bool isptr)
 {
-   TBranch* branch = GetBranch(bname);
-   if (!branch) {
-      if (ptr) *ptr = nullptr;
-      Error("SetBranchAddress", "unknown branch -> %s", bname);
-      return kMissingBranch;
-   }
+   if (auto *branchFromSelf = GetBranchFromSelf(bname)) {
+      Int_t res = CheckBranchAddressType(branchFromSelf, ptrClass, datatype, isptr);
 
-   Int_t res = CheckBranchAddressType(branch, ptrClass, datatype, isptr);
+      // This will set the value of *ptr to branch.
+      if (res >= 0) {
+         // The check succeeded.
+         if ((res & kNeedEnableDecomposedObj) && !branchFromSelf->GetMakeClass())
+            branchFromSelf->SetMakeClass(true);
+         SetBranchAddressImp(branchFromSelf, addr, ptr);
+      } else {
+         if (ptr)
+            *ptr = nullptr;
+      }
+      return res;
+   }
 
-   // This will set the value of *ptr to branch.
-   if (res >= 0) {
-      // The check succeeded.
-      if ((res & kNeedEnableDecomposedObj) && !branch->GetMakeClass())
-         branch->SetMakeClass(true);
-      SetBranchAddressImp(branch,addr,ptr);
-   } else {
-      if (ptr) *ptr = nullptr;
+   // Check friends
+   if (fFriends) {
+      for (auto *fe : TRangeDynCast<TFriendElement>(fFriends)) {
+         if (auto *tree = fe->GetTree()) {
+            int status = tree->SetBranchAddress(bname, addr, ptr);
+            if (status != kMissingBranch) {
+               return status;
+            }
```

Comment:
```suggestion
            return status;
```
Now, that we use the full `SetBranchAddress`, the error handling is already done, isn't it?